subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2637,5357 +1832,233 @@ static void conn_cleanup_iso_cis_released_cb(struct ll_conn *conn)
+	/* Check if any pending LLL events that need to be aborted */
+	mfy.param = param;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_LOW,
+			     TICKER_USER_ID_ULL_HIGH, 0, &mfy);
+	LL_ASSERT(!ret);
+}
 
-			conn->llcp_phy.tx &= p->rx_phys;
-			conn->llcp_phy.rx &= p->tx_phys;
+static void conn_disable(void *param)
+{
+	struct ll_conn *conn;
+	struct ull_hdr *hdr;
 
-			if (!conn->llcp_phy.tx || !conn->llcp_phy.rx) {
-				conn->llcp_phy.tx = 0U;
-				conn->llcp_phy.rx = 0U;
-			}
+	/* Check ref count to determine if any pending LLL events in pipeline */
+	conn = param;
+	hdr = &conn->ull;
+	if (ull_ref_get(hdr)) {
+		static memq_link_t link;
+		static struct mayfly mfy = {0, 0, &link, NULL, lll_disable};
+		uint32_t ret;
 
-			/* pause data packet tx */
-			conn->llcp_phy.pause_tx = 1U;
+		mfy.param = &conn->lll;
 
-			/* Procedure timeout is stopped */
-			conn->procedure_expire = 0U;
-		}
+		/* Setup disabled callback to be called when ref count
+		 * returns to zero.
+		 */
+		LL_ASSERT(!hdr->disabled_cb);
+		hdr->disabled_param = mfy.param;
+		hdr->disabled_cb = disabled_cb;
 
-		/* Mark for buffer for release */
-		(*rx)->hdr.type = NODE_RX_TYPE_RELEASE;
+		/* Trigger LLL disable */
+		ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH,
+				     TICKER_USER_ID_LLL, 0, &mfy);
+		LL_ASSERT(!ret);
+	} else {
+		/* No pending LLL events */
+		disabled_cb(&conn->lll);
+	}
+}
 
-		break;
-#endif /* CONFIG_BT_CENTRAL */
+static void disabled_cb(void *param)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, tx_lll_flush};
+	uint32_t ret;
 
-#if defined(CONFIG_BT_PERIPHERAL)
-	case PDU_DATA_LLCTRL_TYPE_PHY_UPD_IND:
-	{
-		uint8_t err;
+	mfy.param = param;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH,
+			     TICKER_USER_ID_LLL, 0, &mfy);
+	LL_ASSERT(!ret);
+}
 
-		if (!conn->lll.role ||
-		    PDU_DATA_LLCTRL_LEN(phy_upd_ind) != pdu_rx->len) {
-			goto ull_conn_rx_unknown_rsp_send;
-		}
+static void tx_lll_flush(void *param)
+{
+	struct node_rx_pdu *rx;
+	struct lll_conn *lll;
+	struct ll_conn *conn;
+	struct node_tx *tx;
+	memq_link_t *link;
+	uint16_t handle;
 
-		err = phy_upd_ind_recv(conn, link, rx, pdu_rx);
-		if (err) {
-			conn->llcp_terminate.reason_final = err;
-		}
-	}
-	break;
-#endif /* CONFIG_BT_PERIPHERAL */
-#endif /* CONFIG_BT_CTLR_PHY */
+	/* Get reference to ULL context */
+	lll = param;
+	conn = HDR_LLL2ULL(lll);
+	handle = ll_conn_handle_get(conn);
 
-#if defined(CONFIG_BT_CTLR_MIN_USED_CHAN)
-#if defined(CONFIG_BT_CENTRAL)
-	case PDU_DATA_LLCTRL_TYPE_MIN_USED_CHAN_IND:
-		if (conn->lll.role ||
-		    PDU_DATA_LLCTRL_LEN(min_used_chans_ind) != pdu_rx->len) {
-			goto ull_conn_rx_unknown_rsp_send;
-		}
+	lll_conn_flush(handle, lll);
 
-		if (!conn->lll.role) {
-			struct pdu_data_llctrl_min_used_chans_ind *p =
-				&pdu_rx->llctrl.min_used_chans_ind;
+	link = memq_dequeue(lll->memq_tx.tail, &lll->memq_tx.head,
+			    (void **)&tx);
+	while (link) {
+		uint8_t idx;
+		struct lll_tx *lll_tx2;
```

Comment:
Because one of the checks in CI complains if the variable has the same name as the struct name. Agreed that the suffix 2 does not give a better name though